v3/lints/cabf_br/lint_sub_cert_eku_check.go

```diff
@@ -0,0 +1,83 @@
+package cabf_br
+
+/*
+ * ZLint Copyright 2024 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import (
+	"fmt"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/util"
+)
+
+type subExtKeyUsageCheck struct{}
+
+func init() {
+	lint.RegisterCertificateLint(&lint.CertificateLint{
+		LintMetadata: lint.LintMetadata{
+			Name:          "e_sub_cert_eku_check",
+			Description:   "Subscriber certificates MUST have id-kp-serverAuth and MAY have id-kp-clientAuth present in extKeyUsage",
+			Citation:      "BRs: 7.1.2.7.10 Subscriber Certificate Extended Key Usage",
+			Source:        lint.CABFBaselineRequirements,
+			EffectiveDate: util.CABFBRs_2_0_0_Date,
+		},
+		Lint: NewSubExtKeyUsageCheck,
+	})
+}
+
+func NewSubExtKeyUsageCheck() lint.LintInterface {
+	return &subExtKeyUsageCheck{}
+}
+
+func (l *subExtKeyUsageCheck) CheckApplies(c *x509.Certificate) bool {
+	return c.ExtKeyUsage != nil
```

Comment:
I think this should just be `util.IsSubscriberCert()` or `util.IsSubscriberCert() && util.IsExtInCert(c, util.EkuSynOid)`